src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py

```diff
@@ -75,6 +113,10 @@ def fit(
             less than ensemble_size - 1 and the dynamical model is non-linear.
         """
 
+        # ---------------------------------------------------------------------
```

Comment:
Not a big fan of these large headers.
You like?

Reply:
If they are not overused, I like them! Here the method does two things: (1) housekeeping / input validation and (2) computatations -- and I like splitting it up for the reader.

Reply:
From my experience they usually end up outdated rather quickly, but can leave for now.